### DISCUSSION TITLE
Profiller SQL Fastapi

### DISCUSSION THREAD

**Discussion by hamedyosefian** [fastapi/fastapi#5763] [Questions]
I want show touch database and sql commands in fast api similar django-debug-toolbar

fastapi-debug-toolbar not work for me (not show sqlalchemy querys)

Whats the best practice ?

**Comment by caeser1996** [comment]
There are a few options you can use to show database queries and SQL commands in FastAPI similar to Django Debug Toolbar:

SQLAlchemy Event Listeners: SQLAlchemy provides event listeners that you can use to intercept and log SQL statements before they are executed. You can use this feature to log the queries and SQL commands for your application.

Flask-SQLAlchemy Debug Toolbar: Flask-SQLAlchemy Debug Toolbar is a fork of Django Debug Toolbar that has been adapted for Flask and SQLAlchemy. It provides a similar functionality to Django Debug Toolbar, but with a Flask and SQLAlchemy specific implementation.

Debugging and Logging: If you don't want to use a third-party package, you can use built-in debugging and logging features in Python to log the SQL commands executed by SQLAlchemy. This can be done by setting the log level of your logger to DEBUG, which will log all the SQL statements executed by SQLAlchemy.

These are some of the best practices to show SQL commands in FastAPI applications. You can choose the option that best fits your needs and development style.
